app/Http/Controllers/NilaiController.php

```diff
@@ -90,57 +91,105 @@ public function my_array_unique($array, $keep_key_assoc = false){
+                    ->where('jawaban__tps.modul_id', $modul_id)
+                    ->leftJoin('soal__tps', 'jawaban__tps.soal_id', '=', 'soal__tps.id')
+                    ->get();
+                
+                foreach ($jawabans as $jawaban => $value)
+                    array_push($allJawabanTp, $value);
+    
+                $allJawabanTp = $this->my_array_unique($allJawabanTp);
+    
+            } else {
+                $allJawabanTp = "nope";
+            }
+    
+            return response()->json([
+                'message' => 'success',
+                'allJawabanTp' => $allJawabanTp,
+                'allJawabanJurnal' => $allJawabanJurnal,
+            ], 200);   
+        }
 
-            $allJawabanTp = $this->my_array_unique($allJawabanTp);
+        /* if logged in as praktikan */
+        else if ($isPraktikan) {
+            $jawabans = Jawaban_Fitb::where('praktikan_id', $praktikan_id)
+                ->where('jawaban__fitbs.modul_id', $modul_id)
+                ->leftJoin('soal__fitbs', 'jawaban__fitbs.soal_id', '=', 'soal__fitbs.id')
+                ->join('moduls', 'jawaban__fitbs.modul_id', '=', 'moduls.id')
+                ->where('moduls.isUnlocked', 1)
+                ->select('jawaban__fitbs.jawaban', 'soal__fitbs.soal')
+                ->get();
+            
+            foreach ($jawabans as $jawaban => $value)
+                array_push($allJawabanJurnal, $value);  
+            
+            $jawabans = Jawaban_Jurnal::where('praktikan_id', $praktikan_id)
+                ->where('jawaban__jurnals.modul_id', $modul_id)
+                ->leftJoin('soal__jurnals', 'jawaban__jurnals.soal_id', '=', 'soal__jurnals.id')
+                ->join('moduls', 'jawaban__jurnals.modul_id', '=', 'moduls.id')
+                ->where('moduls.isUnlocked', 1)
+                ->select('jawaban__jurnals.jawaban', 'soal__jurnals.soal')
+                ->get();
+            
+            foreach ($jawabans as $jawaban => $value)
+                array_push($allJawabanJurnal, $value);  
+            
+            $jawabans = Jawaban_Mandiri::where('praktikan_id', $praktikan_id)
+                ->where('jawaban__mandiris.modul_id', $modul_id)
+                ->leftJoin('soal__mandiris', 'jawaban__mandiris.soal_id', '=', 'soal__mandiris.id')
+                ->join('moduls', 'jawaban__mandiris.modul_id', '=', 'moduls.id')
+                ->where('moduls.isUnlocked', 1)
+                ->select('jawaban__mandiris.jawaban', 'soal__mandiris.soal')
+                ->get();
+            
+            foreach ($jawabans as $jawaban => $value)
+                array_push($allJawabanJurnal, $value);
+    
+            $allJawabanJurnal = $this->my_array_unique($allJawabanJurnal);
 
-        } else {
-            $allJawabanTp = "nope";
+            return response()->json([
+                'message' => 'success',
+                'allJawabanJurnal' => $allJawabanJurnal,
+            ], 200);
```

Comment:
hmm i don't know is right or not, i think u'll better to create new `controller file` specified like **Answers Controller, or etc**, why because if u place in here it's ambiguous and first i saw this.. i think this is for show praktikans' score but the fact is for show the pratikans' answers 😅

Reply:
Actually I too confused about which is better between make a new controller or just merge it in an existing controller 😅, If it's better to make a new controller for a better documentation, then I'll make it. Thanks for the review bang Adhi 😁